lib/mongodb.js

```diff
@@ -2023,28 +2033,49 @@ function visitAllProperties(data, modelCtorOrDef, visitor) {
         visitAllProperties(value, def.type.definition, visitor);
       }
     } else {
-      visitor(value, def, (newValue) => { data[p] = newValue; });
+      visitor(model, value, def, (newValue) => { data[p] = newValue; });
     }
     continue;
   }
 }
 
 /**
- *
- * @param {*} propValue Property value to coerce into a Decimal128 value
- * @param {*} propDef Property definition to check if property is MongoDB
- * Decimal128 type
+ * @param {*} model Model containing the property
+ * @param {*} propValue Property value to coerce into special types supported by the connector
+ * @param {*} propDef Property definition to check if property is for MongoDB
  */
-function coerceDecimalProperty(propValue, propDef, setValue) {
-  let updatedValue;
-  if (hasDataType('decimal128', propDef)) {
-    if (Array.isArray(propValue)) {
-      updatedValue = propValue.map(val => Decimal128.fromString(val));
-      return setValue(updatedValue);
-    } else {
-      updatedValue = Decimal128.fromString(propValue);
-      return setValue(updatedValue);
+function coerceProperties(model, propValue, propDef, setValue) {
+  let coercedValue;
+  // Process only mongo-specific data types
+  if (propDef && propDef.mongodb && propDef.mongodb.dataType) {
+    const dataType = propDef.mongodb.dataType;
+    // Data types specified as string values
+    // Yaapa: should we implement all type specifications using objects instead of strings?
```

Comment:
I am not sure what's the actual question here.

Are you asking about MongoDB-specific types other than `ObjectID` and `Decimal128`, e.g. `timestamp` (see [MongoDB's doc on BSON types](https://docs.mongodb.com/manual/reference/bson-types/index.html))? IMO, it would be great eventually support all BSON types, but such changes are definitely out of scope of this pull request.